File: src/local_bot_client.py
import aiohttp
import logging

class LocalBotClient:
    def __init__(self, api_url):
        self.api_url = api_url
        self.conversation_id = 0

    async def get_response(self, message):
        try:
            async with aiohttp.ClientSession() as session:
                async with session.post(
                    self.api_url,
                    json={
                        "conversation_id": self.conversation_id,
                        "content": message,
                        "collection": "general"
                    }
                ) as response:
                    if response.status == 200:
                        data = await response.json()
                        return data["bot_response"]
                    else:
                        logging.error(f"Error from local LLM API: {response.status}")
                        return "Xin lỗi, tôi đang gặp sự cố kỹ thuật."
        except Exception as e:
            logging.error(f"Error calling local LLM API: {e}")
            return "Xin lỗi, tôi đang gặp sự cố kỹ thuật."

    def end_conversation(self):
        self.conversation_id = 0 

File: callbot_go/src/local_bot_client.go
package src

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"log"
	"net/http"
)

type LocalBotClient struct {
	ApiUrl         string
	ConversationId int
}

type RequestBody struct {
	ConversationId int    `json:"conversation_id"`
	Content        string `json:"content"`
	Collection     string `json:"collection"`
}

type ResponseBody struct {
	BotResponse string `json:"bot_response"`
}

func NewLocalBotClient(apiUrl string) *LocalBotClient {
	return &LocalBotClient{
		ApiUrl:         apiUrl,
		ConversationId: 0,
	}
}

func (c *LocalBotClient) GetResponse(message string) (string, error) {
	reqBody := RequestBody{
		ConversationId: c.ConversationId,
		Content:        message,
		Collection:     "general",
	}

	jsonData, err := json.Marshal(reqBody)
	if err != nil {
		log.Printf("Error marshaling request: %v", err)
		return "Xin lỗi, tôi đang gặp sự cố kỹ thuật.", err
	}

	resp, err := http.Post(c.ApiUrl, "application/json", bytes.NewBuffer(jsonData))
	if err != nil {
		log.Printf("Error calling local LLM API: %v", err)
		return "Xin lỗi, tôi đang gặp sự cố kỹ thuật.", err
	}
	defer resp.Body.Close()

	if resp.StatusCode == http.StatusOK {
		body, err := io.ReadAll(resp.Body)
		if err != nil {
			log.Printf("Error reading response body: %v", err)
			return "Xin lỗi, tôi đang gặp sự cố kỹ thuật.", err
		}

		var response ResponseBody
		if err := json.Unmarshal(body, &response); err != nil {
			log.Printf("Error unmarshaling response: %v", err)
			return "Xin lỗi, tôi đang gặp sự cố kỹ thuật.", err
		}

		return response.BotResponse, nil
	}

	log.Printf("Error from local LLM API: %d", resp.StatusCode)
	return "Xin lỗi, tôi đang gặp sự cố kỹ thuật.", fmt.Errorf("API returned status code %d", resp.StatusCode)
}

func (c *LocalBotClient) EndConversation() {
	c.ConversationId = 0
}
